applicationset/generators/cluster.go

```diff
@@ -139,6 +151,16 @@ func (g *ClusterGenerator) GenerateParams(
 	return res, nil
 }
 
+func replaceTemplatedString(value string, params map[string]string, key string) error {
+	fstTmpl := fasttemplate.New(value, "{{", "}}")
+	replacedTmplStr, err := render.Replace(fstTmpl, params, true)
+	if err != nil {
+		return err
+	}
+	params[fmt.Sprintf("values.%s", key)] = replacedTmplStr
```

Comment:
I'm a little worried about setting directly to `params`. At this point, AppSet is admin-only, but in the future I'd like it to see it made available to lower-privileged users. In that case, I'd be concerned about a billion-laughs attack.

The attack would work like this:

```yaml
values:
  lol1: lol
  lol2: '{{values.lol1}}{{values.lol1}}{{values.lol1}}{{values.lol1}}{{values.lol1}}{{values.lol1}}{{values.lol1}}{{values.lol1}}{{values.lol1}}{{values.lol1}}'
  lol3: '{{values.lol2}}{{values.lol2}}{{values.lol2}}{{values.lol2}}{{values.lol2}}{{values.lol2}}{{values.lol2}}{{values.lol2}}{{values.lol2}}{{values.lol2}}'
  # ...etc. through lol9
```

There are probably other/easier ways to attack the AppSet controller, but I'd rather not add another one.

What if we just keep track of an `interpolatedParams` map which isn't merged back with `params` until all the other interpolation is finished?

Reply:
> What if we just keep track of an interpolatedParams map which isn't merged back with params until all the other interpolation is finished?

Ah, good catch! I updated the PR to do just that, and added a test case for it as well.